src/BBT.StructureTools.Tests/Compare/ComparerWithObjectInheritingFromBaseModelAttributeWithDistinguishedComparerTests.cs

```diff
@@ -0,0 +1,231 @@
+
+            // Assert
+            result.Should().BeTrue();
+        }
+
+        /// <summary>
+        /// Tests IComparer.Equals.
+        /// </summary>
+        [Fact]
+        public void Equals_WhenBaseModelAttributeObjectsNotEqualAndHaveDifferentValue_MustReturnFalse()
+        {
+            // Arrange
+            var baseModelTestAttribute = new BaseModelTestAttribute { TestValue1 = 55 };
+            var baseModelTestAttribute2 = new BaseModelTestAttribute();
+            var testClassA = new TestClass { BaseModelTestAttribute = baseModelTestAttribute };
+            var testClassB = new TestClass { BaseModelTestAttribute = baseModelTestAttribute2 };
+
+            // Act
+            var result = this.comparer.Equals(testClassA, testClassB);
+
+            // Assert
+            result.Should().BeFalse();
+        }
+
+        /// <summary>
+        /// Tests IComparer.Equals.
+        /// </summary>
+        [Fact]
+        public void Equals_WhenAttributeObjectsNotEqualButNotRegistered_MustReturnTrue()
+        {
+            // Arrange
+            var baseModelTestAttribute = new BaseModelTestAttribute() { TestValue2 = 2 };
+            var baseModelTestAttribute2 = new BaseModelTestAttribute() { TestValue2 = 1 };
+            var testClassA = new TestClass { BaseModelTestAttribute = baseModelTestAttribute };
+            var testClassB = new TestClass { BaseModelTestAttribute = baseModelTestAttribute2 };
+
+            // Act
+            var result = this.comparer.Equals(testClassA, testClassB);
+
+            // Assert
+            result.Should().BeTrue();
+        }
+
+        /// <summary>
+        /// Tests IComparer.Equals.
+        /// </summary>
+        [Fact]
+        public void Equals_WhenAttributeObjectsNotEqualButExcluded_MustReturnTrue()
+        {
+            // Arrange
+            var baseModelTestAttribute = new BaseModelTestAttribute() { TestValue1 = 2 };
+            var baseModelTestAttribute2 = new BaseModelTestAttribute() { TestValue1 = 1 };
+            var testClassA = new TestClass { BaseModelTestAttribute = baseModelTestAttribute };
+            var testClassB = new TestClass { BaseModelTestAttribute = baseModelTestAttribute2 };
+            var comparerExclusions = new List<IComparerExclusion>
+                                          {
+                                              new PropertyComparerExclusion<TestClass>(x => x.BaseModelTestAttribute),
+                                          };
+
+            // Act
+            var result = this.comparer.Equals(testClassA, testClassB, Array.Empty<IBaseAdditionalProcessing>(), comparerExclusions);
+
+            // Assert
+            result.Should().BeTrue();
+        }
+
+        #endregion
+
+        #region private test classes and test class helpers
+        private class TestClass
+        {
+            public BaseModelTestAttribute BaseModelTestAttribute { get; set; }
+        }
+
+        private class BaseModelTestAttribute
```

Comment:
was inheriting `BaseModel, IBaseModel` -> that's why we should keep this test @BBT

Reply:
ditto